tket/tests/ZX/test_ZXConverters.cpp

```diff
@@ -0,0 +1,664 @@
+// Copyright 2019-2022 Cambridge Quantum Computing
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <boost/graph/adjacency_list.hpp>
+#include <catch2/catch.hpp>
+
+#include "../testutil.hpp"
+#include "Converters/Converters.hpp"
+#include "Gate/SymTable.hpp"
+namespace tket {
+namespace zx {
+namespace test_ZXConverters {
+SCENARIO("Check converting gates to spiders") {
+  GIVEN("X") {
+    Circuit circ(1);
+    circ.add_op<unsigned>(OpType::X, {0});
+    ZXDiagram zx;
+    boost::bimap<ZXVert, Vertex> bmap;
+    std::tie(zx, bmap) = circuit_to_zx(circ);
+    ZXVertVec boundary = zx.get_boundary(ZXType::Input, QuantumType::Quantum);
+    ZXVert input = boundary[0];
+    ZXVert x = zx.neighbours(input)[0];
+    PhasedGen x_gen = zx.get_vertex_ZXGen<PhasedGen>(x);
+    REQUIRE(x_gen.get_type() == ZXType::XSpider);
```

Comment:
I'd generally recommend checking the `ZXType` before casting to a specific generator subclass to prevent errors from bad casts, even during tests where we expect it to be a known `ZXType`